File: plus-one.rb
# @param {Integer[]} digits
# @return {Integer[]}
def plus_one(digits)
  ret = []
  carry = true
  for i in 0...digits.size
    d = digits[digits.size - 1 - i]
    if carry
      d = d + 1
      carry = false
    end
    if d >= 10
      carry = true
      d = d - 10
    end
    ret.push(d)
  end
  if carry
    ret.push(1)
  end
  return ret.reverse
end

File: plus-one.swift

class Solution {
    func plusOne(digits: [Int]) -> [Int] {
        var carry = 1
        var result: [Int] = []
        for digit in digits.reverse() {
            var r = digit + carry
            if r == 10 {
               r -= 10 
               carry = 1
            } else {
                carry = 0
            }
            result.append(r)
        }
        if carry == 1 {
            result.append(1)
        }
        return result.reverse()
    }
}
